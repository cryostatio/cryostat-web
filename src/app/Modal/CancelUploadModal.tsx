/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { portalRoot } from '@app/utils/utils';
import { Button, Modal } from '@patternfly/react-core';
import * as React from 'react';

export interface CancelUploadModalProps {
  visible: boolean;
  onYes: () => void;
  onNo: () => void;
  title: string;
  message: string;
}

export const CancelUploadModal: React.FC<CancelUploadModalProps> = ({ visible, onYes, onNo, title, message }) => {
  return (
    <Modal
      appendTo={portalRoot}
      width={'40%'}
      isOpen={visible}
      showClose={true}
      onClose={onNo}
      title={title}
      actions={[
        <Button key={'Yes'} variant="primary" onClick={onYes}>
          Yes
        </Button>,
        <Button key={'No'} variant="secondary" onClick={onNo}>
          No
        </Button>,
      ]}
    >
      {message}
    </Modal>
  );
};
