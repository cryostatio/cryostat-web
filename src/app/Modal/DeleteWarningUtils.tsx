/*
 * Copyright The Cryostat Authors
 *
 * The Universal Permissive License (UPL), Version 1.0
 *
 * Subject to the condition set forth below, permission is hereby granted to any
 * person obtaining a copy of this software, associated documentation and/or data
 * (collectively the "Software"), free of charge and under any and all copyright
 * rights in the Software, and any and all patent rights owned or freely
 * licensable by each licensor hereunder covering either (i) the unmodified
 * Software as contributed to or provided by such licensor, or (ii) the Larger
 * Works (as defined below), to deal in both
 *
 * (a) the Software, and
 * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
 * one is included with the Software (each a "Larger Work" to which the Software
 * is contributed by such licensors),
 *
 * without restriction, including without limitation the rights to copy, create
 * derivative works of, display, perform, and distribute the Software and make,
 * use, sell, offer for sale, import, export, have made, and have sold the
 * Software and the Larger Work(s), and to sublicense the foregoing rights on
 * either these or other terms.
 *
 * This license is subject to the following condition:
 * The above copyright notice and either this complete permission notice or at
 * a minimum a reference to the UPL must be included in all copies or
 * substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
export enum DeleteWarningType {
  DeleteActiveRecordings = 'DeleteActiveRecordings',
  DeleteArchivedRecordings = 'DeleteArchivedRecordings',
  DeleteAutomatedRules = 'DeleteAutomatedRules',
  DeleteEventTemplates = 'DeleteEventTemplates',
  DeleteProbeTemplates = 'DeleteProbeTemplates',
  DeleteActiveProbes = 'DeleteActiveProbes',
  DeleteJMXCredentials = 'DeleteJMXCredentials',
  DeleteCustomTargets = 'DeleteCustomTargets',
}

export interface DeleteWarning {
  id: DeleteWarningType;
  title: string;
  label: string;
  description: string;
  ariaLabel: string;
}

export const DeleteActiveRecordings: DeleteWarning = {
  id: DeleteWarningType.DeleteActiveRecordings,
  title: 'Permanently delete Active Recording?',
  label: 'Delete Active Recording',
  description: `Recording and report data will be lost.`,
  ariaLabel: 'Recording delete warning',
};

export const DeleteArchivedRecordings: DeleteWarning = {
  id: DeleteWarningType.DeleteArchivedRecordings,
  title: 'Permanently delete Archived Recording?',
  label: 'Delete Archived Recording',
  description: `Recording and report data will be lost.`,
  ariaLabel: 'Recording delete warning',
};

export const DeleteAutomatedRules: DeleteWarning = {
  id: DeleteWarningType.DeleteAutomatedRules,
  title: 'Permanently delete Automated Rule?',
  label: 'Delete Automated Rule',
  description: `Rule data will be lost.`,
  ariaLabel: 'Automated rule delete warning',
};

export const DeleteEventTemplates: DeleteWarning = {
  id: DeleteWarningType.DeleteEventTemplates,
  title: 'Permanently delete Event Template?',
  label: 'Delete Event Template',
  description: `Custom event template data will be lost.`,
  ariaLabel: 'Event template delete warning',
};

export const DeleteProbeTemplates: DeleteWarning = {
  id: DeleteWarningType.DeleteProbeTemplates,
  title: 'Permanently delete Probe Template?',
  label: 'Delete Probe Template',
  description: `Custom probe template data will be lost.`,
  ariaLabel: 'Probe template delete warning',
};

export const DeleteActiveProbes: DeleteWarning = {
  id: DeleteWarningType.DeleteActiveProbes,
  title: 'Permanently remove Active Probes from the target?',
  label: 'Remove Active Probes',
  description: `Active probes will be removed from the target.`,
  ariaLabel: 'Active Probes remove warning',
};

export const DeleteJMXCredentials: DeleteWarning = {
  id: DeleteWarningType.DeleteJMXCredentials,
  title: 'Permanently delete JMX Credentials?',
  label: 'Delete JMX Credentials',
  description: `Credential data for this target will be lost.`,
  ariaLabel: 'JMX Credentials delete warning',
};

export const DeleteCustomTargets: DeleteWarning = {
  id: DeleteWarningType.DeleteCustomTargets,
  title: 'Permanently delete Custom Target?',
  label: 'Delete Custom Targets',
  description: `Custom target information will be lost.`,
  ariaLabel: 'Custom Targets delete warning',
};

export const DeleteWarningKinds: DeleteWarning[] = [
  DeleteActiveRecordings,
  DeleteArchivedRecordings,
  DeleteAutomatedRules,
  DeleteEventTemplates,
  DeleteProbeTemplates,
  DeleteActiveProbes,
  DeleteJMXCredentials,
  DeleteCustomTargets,
];

export const getFromWarningMap = (warning: DeleteWarningType): DeleteWarning | undefined => {
  const wt = DeleteWarningKinds.find((t) => t.id === warning);
  return wt;
};
