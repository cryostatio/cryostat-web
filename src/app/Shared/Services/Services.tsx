/*
 * Copyright The Cryostat Authors
 *
 * The Universal Permissive License (UPL), Version 1.0
 *
 * Subject to the condition set forth below, permission is hereby granted to any
 * person obtaining a copy of this software, associated documentation and/or data
 * (collectively the "Software"), free of charge and under any and all copyright
 * rights in the Software, and any and all patent rights owned or freely
 * licensable by each licensor hereunder covering either (i) the unmodified
 * Software as contributed to or provided by such licensor, or (ii) the Larger
 * Works (as defined below), to deal in both
 *
 * (a) the Software, and
 * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
 * one is included with the Software (each a "Larger Work" to which the Software
 * is contributed by such licensors),
 *
 * without restriction, including without limitation the rights to copy, create
 * derivative works of, display, perform, and distribute the Software and make,
 * use, sell, offer for sale, import, export, have made, and have sold the
 * Software and the Larger Work(s), and to sublicense the foregoing rights on
 * either these or other terms.
 *
 * This license is subject to the following condition:
 * The above copyright notice and either this complete permission notice or at
 * a minimum a reference to the UPL must be included in all copies or
 * substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
import * as React from 'react';
import { NotificationsInstance } from '@app/Notifications/Notifications';
import { TargetService, TargetInstance } from './Target.service';
import { TargetsService } from './Targets.service';
import { ApiService } from './Api.service';
import { NotificationChannel } from './NotificationChannel.service';
import { ReportService } from './Report.service';
import { SettingsService } from './Settings.service';
import { LoginService } from './Login.service';
import { JmxCredentials } from './JmxCredentials.service';

export interface Services {
  target: TargetService;
  targets: TargetsService;
  api: ApiService;
  jmxCredentials: JmxCredentials;
  notificationChannel: NotificationChannel;
  reports: ReportService;
  settings: SettingsService;
  login: LoginService;
}

const settings = new SettingsService();
const jmxCredentials = new JmxCredentials(() => api);
const login = new LoginService(TargetInstance, jmxCredentials, settings);
const api = new ApiService(TargetInstance, NotificationsInstance, login);
const notificationChannel = new NotificationChannel(NotificationsInstance, login);
const reports = new ReportService(login, NotificationsInstance);
const targets = new TargetsService(api, NotificationsInstance, login, notificationChannel);

const defaultServices: Services = {
  target: TargetInstance,
  targets,
  api,
  jmxCredentials,
  notificationChannel,
  reports,
  settings,
  login,
};

const ServiceContext: React.Context<Services> = React.createContext(defaultServices);

export { ServiceContext, defaultServices };
