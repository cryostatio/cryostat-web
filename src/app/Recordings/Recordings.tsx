/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ServiceContext } from '@app/Shared/Services/Services';
import { TargetView } from '@app/TargetView/TargetView';
import { useSubscriptions } from '@app/utils/useSubscriptions';
import { getActiveTab, switchTab } from '@app/utils/utils';
import { Card, CardBody, CardTitle, Tab, Tabs, TabTitleText } from '@patternfly/react-core';
import * as React from 'react';
import { useHistory, useLocation } from 'react-router';
import { ActiveRecordingsTable } from './ActiveRecordingsTable';
import { ArchivedRecordingsTable } from './ArchivedRecordingsTable';

enum RecordingTab {
  ACTIVE_RECORDING = 'active-recording',
  ARCHIVED_RECORDING = 'archived-recording',
}

export interface RecordingsProps {}

export const Recordings: React.FC<RecordingsProps> = ({ ...props }) => {
  const { search, pathname } = useLocation();
  const history = useHistory();
  const context = React.useContext(ServiceContext);
  const addSubscription = useSubscriptions();

  const activeTab = React.useMemo(() => {
    return getActiveTab(search, 'tab', Object.values(RecordingTab), RecordingTab.ACTIVE_RECORDING);
  }, [search]);

  const [archiveEnabled, setArchiveEnabled] = React.useState(false);

  React.useEffect(() => {
    addSubscription(context.api.isArchiveEnabled().subscribe(setArchiveEnabled));
  }, [context.api, addSubscription, setArchiveEnabled]);

  const onTabSelect = React.useCallback(
    (_: React.MouseEvent, key: string | number) =>
      switchTab(history, pathname, search, { tabKey: 'tab', tabValue: `${key}` }),
    [history, pathname, search],
  );

  const targetAsObs = React.useMemo(() => context.target.target(), [context.target]);

  const cardBody = React.useMemo(() => {
    return archiveEnabled ? (
      <Tabs id="recordings" activeKey={activeTab} onSelect={onTabSelect} unmountOnExit>
        <Tab
          id="active-recordings"
          eventKey={RecordingTab.ACTIVE_RECORDING}
          title={<TabTitleText>Active Recordings</TabTitleText>}
          data-quickstart-id="active-recordings-tab"
        >
          <ActiveRecordingsTable archiveEnabled={true} />
        </Tab>
        <Tab
          id="archived-recordings"
          eventKey={RecordingTab.ARCHIVED_RECORDING}
          title={<TabTitleText>Archived Recordings</TabTitleText>}
          data-quickstart-id="archived-recordings-tab"
        >
          <ArchivedRecordingsTable target={targetAsObs} isUploadsTable={false} isNestedTable={false} />
        </Tab>
      </Tabs>
    ) : (
      <>
        <CardTitle>Active Recordings</CardTitle>
        <ActiveRecordingsTable archiveEnabled={false} />
      </>
    );
  }, [archiveEnabled, activeTab, onTabSelect, targetAsObs]);

  return (
    <TargetView {...props} pageTitle="Recordings">
      <Card>
        <CardBody>{cardBody}</CardBody>
      </Card>
    </TargetView>
  );
};

export default Recordings;
