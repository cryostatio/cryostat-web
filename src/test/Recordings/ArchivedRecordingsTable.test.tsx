/*
 * Copyright The Cryostat Authors
 *
 * The Universal Permissive License (UPL), Version 1.0
 *
 * Subject to the condition set forth below, permission is hereby granted to any
 * person obtaining a copy of this software, associated documentation and/or data
 * (collectively the "Software"), free of charge and under any and all copyright
 * rights in the Software, and any and all patent rights owned or freely
 * licensable by each licensor hereunder covering either (i) the unmodified
 * Software as contributed to or provided by such licensor, or (ii) the Larger
 * Works (as defined below), to deal in both
 *
 * (a) the Software, and
 * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
 * one is included with the Software (each a "Larger Work" to which the Software
 * is contributed by such licensors),
 *
 * without restriction, including without limitation the rights to copy, create
 * derivative works of, display, perform, and distribute the Software and make,
 * use, sell, offer for sale, import, export, have made, and have sold the
 * Software and the Larger Work(s), and to sublicense the foregoing rights on
 * either these or other terms.
 *
 * This license is subject to the following condition:
 * The above copyright notice and either this complete permission notice or at
 * a minimum a reference to the UPL must be included in all copies or
 * substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
import * as React from 'react';
import renderer, { act } from 'react-test-renderer';
import { render, screen, within } from '@testing-library/react';
import '@testing-library/jest-dom';
import { of } from 'rxjs';
import userEvent from '@testing-library/user-event';
import { ArchivedRecording } from '@app/Shared/Services/Api.service';
import { NotificationMessage } from '@app/Shared/Services/NotificationChannel.service';

const mockConnectUrl = 'service:jmx:rmi://someUrl';
const mockTarget = { connectUrl: mockConnectUrl, alias: 'fooTarget' };
const mockRecordingLabels = {
  someLabel: 'someValue',
};
const mockRecording: ArchivedRecording = {
  name: 'someRecording',
  downloadUrl: 'http://downloadUrl',
  reportUrl: 'http://reportUrl',
  metadata: { labels: mockRecordingLabels },
};
const mockArchivedRecordingsResponse = {
  data: {
    targetNodes: [
      {
        recordings: {
          archived: [mockRecording] as ArchivedRecording[],
        },
      },
    ],
  },
};
const mockAnotherRecording = { ...mockRecording, name: 'anotherRecording' };
const mockCreateNotification = {
  message: { target: mockConnectUrl, recording: mockAnotherRecording },
} as NotificationMessage;
const mockLabelsNotification = {
  message: {
    target: mockConnectUrl,
    recordingName: 'someRecording',
    metadata: { labels: { someLabel: 'someUpdatedValue' } },
  },
} as NotificationMessage;
const mockDeleteNotification = { message: { target: mockConnectUrl, recording: mockRecording } } as NotificationMessage;

jest.mock('@app/Recordings/RecordingFilters', () => {
  return {
    ...jest.requireActual('@app/Recordings/RecordingFilters'),
    RecordingFilters: jest.fn(() => {
      return <div>
                RecordingFilters
             </div>
    })
  };
});

import { ArchivedRecordingsTable } from '@app/Recordings/ArchivedRecordingsTable';
import { ServiceContext, defaultServices } from '@app/Shared/Services/Services';
import { DeleteActiveRecordings, DeleteArchivedRecordings, DeleteWarningType } from '@app/Modal/DeleteWarningUtils';

jest.spyOn(defaultServices.api, 'deleteArchivedRecording').mockReturnValue(of(true));
jest.spyOn(defaultServices.api, 'downloadRecording').mockReturnValue();
jest.spyOn(defaultServices.api, 'downloadReport').mockReturnValue();
jest.spyOn(defaultServices.api, 'grafanaDatasourceUrl').mockReturnValue(of('/grafanaUrl'));
jest.spyOn(defaultServices.api, 'graphql').mockReturnValue(of(mockArchivedRecordingsResponse));
jest.spyOn(defaultServices.api, 'uploadArchivedRecordingToGrafana').mockReturnValue(of(true));

jest.spyOn(defaultServices.target, 'target').mockReturnValue(of(mockTarget));

jest.spyOn(defaultServices.settings, 'deletionDialogsEnabledFor')
  .mockReturnValueOnce(true);

jest
  .spyOn(defaultServices.notificationChannel, 'messages')
  .mockReturnValueOnce(of()) // renders correctly
  .mockReturnValueOnce(of())
  .mockReturnValueOnce(of())
  .mockReturnValueOnce(of())

  .mockReturnValueOnce(of(mockCreateNotification)) // adds a recording table after receiving a notification
  .mockReturnValueOnce(of())
  .mockReturnValueOnce(of())
  .mockReturnValueOnce(of())

  .mockReturnValueOnce(of()) // updates the recording labels after receiving a notification
  .mockReturnValueOnce(of())
  .mockReturnValueOnce(of())
  .mockReturnValueOnce(of(mockLabelsNotification))

  .mockReturnValueOnce(of()) // removes a recording after receiving a notification
  .mockReturnValueOnce(of())
  .mockReturnValueOnce(of(mockDeleteNotification))
  .mockReturnValue(of()); // all other tests

describe('<ArchivedRecordingsTable />', () => {
  it('renders correctly', async () => {
    let tree;
    await act(async () => {
      tree = renderer.create(
        <ServiceContext.Provider value={defaultServices}>
          <ArchivedRecordingsTable />
        </ServiceContext.Provider>
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('adds a recording after receiving a notification', () => {
    render(
      <ServiceContext.Provider value={defaultServices}>
        <ArchivedRecordingsTable />
      </ServiceContext.Provider>
    );
    expect(screen.getByText('someRecording')).toBeInTheDocument();
    expect(screen.getByText('anotherRecording')).toBeInTheDocument();
  });

  it('updates the recording labels after receiving a notification', () => {
    render(
      <ServiceContext.Provider value={defaultServices}>
        <ArchivedRecordingsTable />
      </ServiceContext.Provider>
    );
    expect(screen.getByText('someLabel: someUpdatedValue')).toBeInTheDocument();
    expect(screen.queryByText('someLabel: someValue')).not.toBeInTheDocument();
  });

  it('removes a recording after receiving a notification', () => {
    render(
      <ServiceContext.Provider value={defaultServices}>
        <ArchivedRecordingsTable />
      </ServiceContext.Provider>
    );
    expect(screen.queryByText('someRecording')).not.toBeInTheDocument();
  });

  it('displays the toolbar buttons', () => {
    render(
      <ServiceContext.Provider value={defaultServices}>
        <ArchivedRecordingsTable />
      </ServiceContext.Provider>
    );

    expect(screen.getByText('Delete')).toBeInTheDocument();
    expect(screen.getByText('Edit Labels')).toBeInTheDocument();
  });

  it('opens the labels drawer when Edit Labels is clicked', () => {
    render(
      <ServiceContext.Provider value={defaultServices}>
        <ArchivedRecordingsTable />
      </ServiceContext.Provider>
    );

    const checkboxes = screen.getAllByRole('checkbox');
    const selectAllCheck = checkboxes[0];
    userEvent.click(selectAllCheck);
    userEvent.click(screen.getByText('Edit Labels'));
    expect(screen.getByText('Edit Recording Labels')).toBeInTheDocument();
  });

  it('shows a popup when Delete is clicked and then deletes the recording after clicking confirmation Delete', () => {
    render(
      <ServiceContext.Provider value={defaultServices}>
        <ArchivedRecordingsTable />
      </ServiceContext.Provider>
    );

    const checkboxes = screen.getAllByRole('checkbox');
    const selectAllCheck = checkboxes[0];
    userEvent.click(selectAllCheck);
    userEvent.click(screen.getByText('Delete'));

    expect(screen.getByLabelText(DeleteArchivedRecordings.ariaLabel));

    const deleteRequestSpy = jest.spyOn(defaultServices.api, 'deleteArchivedRecording');
    const dialogWarningSpy = jest.spyOn(defaultServices.settings, 'setDeletionDialogsEnabledFor');
    userEvent.click(screen.getByLabelText("Don't ask me again"));
    userEvent.click(within(screen.getByLabelText(DeleteArchivedRecordings.ariaLabel)).getByText('Delete'));

    expect(deleteRequestSpy).toHaveBeenCalledTimes(1);
    expect(deleteRequestSpy).toBeCalledWith('someRecording');
    expect(dialogWarningSpy).toBeCalledTimes(1);
    expect(dialogWarningSpy).toBeCalledWith(DeleteWarningType.DeleteArchivedRecordings, false);
  });

  it('deletes the recording when Delete is clicked w/o popup warning', () => {
    render(
      <ServiceContext.Provider value={defaultServices}>
        <ArchivedRecordingsTable />
      </ServiceContext.Provider>
    );

    const checkboxes = screen.getAllByRole('checkbox');
    const selectAllCheck = checkboxes[0];
    userEvent.click(selectAllCheck);
    userEvent.click(screen.getByText('Delete'));

    const deleteRequestSpy = jest.spyOn(defaultServices.api, 'deleteArchivedRecording');

    expect(screen.queryByLabelText(DeleteActiveRecordings.ariaLabel)).not.toBeInTheDocument();
    expect(deleteRequestSpy).toHaveBeenCalledTimes(1);
    expect(deleteRequestSpy).toBeCalledWith('someRecording');
  });

  it('downloads a recording when Download Recording is clicked', () => {
    render(
      <ServiceContext.Provider value={defaultServices}>
        <ArchivedRecordingsTable />
      </ServiceContext.Provider>
    );

    userEvent.click(screen.getByLabelText('Actions'));
    userEvent.click(screen.getByText('Download Recording'));

    const downloadRequestSpy = jest.spyOn(defaultServices.api, 'downloadRecording');

    expect(downloadRequestSpy).toHaveBeenCalledTimes(1);
    expect(downloadRequestSpy).toBeCalledWith(mockRecording);
  });

  it('displays the automated analysis report when View Report is clicked', () => {
    render(
      <ServiceContext.Provider value={defaultServices}>
        <ArchivedRecordingsTable />
      </ServiceContext.Provider>
    );

    userEvent.click(screen.getByLabelText('Actions'));
    userEvent.click(screen.getByText('View Report ...'));

    const reportRequestSpy = jest.spyOn(defaultServices.api, 'downloadReport');

    expect(reportRequestSpy).toHaveBeenCalledTimes(1);
  });

  it('uploads a recording to Grafana when View in Grafana is clicked', () => {
    render(
      <ServiceContext.Provider value={defaultServices}>
        <ArchivedRecordingsTable />
      </ServiceContext.Provider>
    );

    userEvent.click(screen.getByLabelText('Actions'));
    userEvent.click(screen.getByText('View in Grafana ...'));

    const grafanaUploadSpy = jest.spyOn(defaultServices.api, 'uploadArchivedRecordingToGrafana');

    expect(grafanaUploadSpy).toHaveBeenCalledTimes(1);
    expect(grafanaUploadSpy).toBeCalledWith('someRecording');
  });
});
