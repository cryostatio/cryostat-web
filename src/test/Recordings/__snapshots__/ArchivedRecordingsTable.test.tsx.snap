// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArchivedRecordingsTable /> renders correctly 1`] = `
<div
  className="pf-c-drawer pf-m-inline"
>
  <div
    className="pf-c-drawer__main"
  >
    <div
      className="pf-c-drawer__content recordings-table-drawer-content"
    >
      <div
        className="pf-c-drawer__body pf-m-padding"
      >
        <div
          className="pf-c-toolbar"
          data-ouia-component-id="OUIA-Generated-Toolbar-3"
          data-ouia-component-type="PF4/Toolbar"
          data-ouia-safe={true}
          id="archived-recordings-toolbar"
        >
          <div
            className="pf-c-toolbar__content"
          >
            <div
              className="pf-c-toolbar__content-section"
            >
              <div>
                RecordingFilters
              </div>
              <div
                className="pf-c-toolbar__group pf-m-button-group"
              >
                <div
                  className="pf-c-toolbar__item"
                >
                  <button
                    aria-disabled={true}
                    aria-label={null}
                    className="pf-c-button pf-m-secondary pf-m-disabled"
                    data-ouia-component-id="OUIA-Generated-Button-secondary-3"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe={true}
                    disabled={true}
                    onClick={[Function]}
                    role={null}
                    tabIndex={null}
                    type="button"
                  >
                    Edit Labels
                  </button>
                </div>
                <div
                  className="pf-c-toolbar__item"
                >
                  <button
                    aria-disabled={true}
                    aria-label={null}
                    className="pf-c-button pf-m-danger pf-m-disabled"
                    data-ouia-component-id="OUIA-Generated-Button-danger-3"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe={true}
                    disabled={true}
                    onClick={[Function]}
                    role={null}
                    tabIndex={null}
                    type="button"
                  >
                    Delete
                  </button>
                </div>
              </div>
            </div>
            <div
              className="pf-c-toolbar__expandable-content"
              id="archived-recordings-toolbar-expandable-content-2"
            >
              <div
                className="pf-c-toolbar__group"
              />
            </div>
          </div>
          <div
            className="pf-c-toolbar__content pf-m-hidden"
            hidden={true}
          >
            <div
              className="pf-c-toolbar__group"
            />
          </div>
        </div>
        <table
          aria-label="Archived Flight Recordings"
          className="pf-c-table pf-m-grid-md"
          data-ouia-component-id="OUIA-Generated-Table-1"
          data-ouia-component-type="PF4/Table"
          data-ouia-safe={true}
          role="grid"
        >
          <thead
            className=""
          >
            <tr
              className=""
              data-ouia-component-id="OUIA-Generated-TableRow-1"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe={true}
              hidden={false}
            >
              <td
                className="pf-c-table__check"
                onMouseEnter={[Function]}
                scope={null}
              >
                <input
                  aria-label="Select all rows"
                  checked={false}
                  name="check-all"
                  onChange={[Function]}
                  type="checkbox"
                />
              </td>
              <th
                className=""
                onMouseEnter={[Function]}
                scope={null}
              />
              <th
                className=""
                onMouseEnter={[Function]}
                scope="col"
              >
                Name
              </th>
              <th
                className=""
                onMouseEnter={[Function]}
                scope="col"
              >
                Labels
              </th>
              <th
                className=""
                onMouseEnter={[Function]}
                scope={null}
              />
            </tr>
          </thead>
          <tbody
            className=""
            role="rowgroup"
          >
            <tr
              className=""
              data-ouia-component-id="OUIA-Generated-TableRow-2"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe={true}
              hidden={false}
            >
              <td
                className=""
              >
                <div
                  className="pf-c-check pf-m-standalone"
                >
                  <input
                    aria-invalid={false}
                    checked={false}
                    className="pf-c-check__input"
                    data-ouia-component-id="OUIA-Generated-Checkbox-1"
                    data-ouia-component-type="PF4/Checkbox"
                    data-ouia-safe={true}
                    disabled={false}
                    id="archived-table-row-0-check"
                    name="archived-table-row-0-check"
                    onChange={[Function]}
                    type="checkbox"
                  />
                </div>
              </td>
              <td
                aria-controls="archived-ex-expand-0"
                className="pf-c-table__toggle"
                id="archived-ex-toggle-0"
              >
                <button
                  aria-disabled={false}
                  aria-expanded={false}
                  aria-label="Details"
                  aria-labelledby="simple-node0 expand-toggle0"
                  className="pf-c-button pf-m-plain"
                  data-ouia-component-id="OUIA-Generated-Button-plain-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe={true}
                  disabled={false}
                  id="expand-toggle0"
                  onClick={[Function]}
                  role={null}
                  type="button"
                >
                  <div
                    className="pf-c-table__toggle-icon"
                  >
                    <svg
                      aria-hidden={true}
                      aria-labelledby={null}
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style={
                        Object {
                          "verticalAlign": "-0.125em",
                        }
                      }
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                      />
                    </svg>
                  </div>
                </button>
              </td>
              <td
                className=""
                data-label="Name"
              >
                someRecording
              </td>
              <td
                className=""
                data-label="Labels"
              >
                <span
                  className="pf-c-label"
                >
                  <span
                    className="pf-c-label__content"
                  >
                    someLabel: someValue
                  </span>
                </span>
              </td>
              <td
                className="pf-c-table__action"
                style={
                  Object {
                    "paddingRight": 0,
                    "width": "auto",
                  }
                }
              >
                <div
                  className="pf-c-dropdown pf-m-align-right"
                  data-ouia-component-id="OUIA-Generated-Dropdown-1"
                  data-ouia-component-type="PF4/Dropdown"
                  data-ouia-safe={true}
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    aria-label="Actions"
                    className="pf-c-dropdown__toggle pf-m-plain"
                    disabled={false}
                    id="pf-dropdown-toggle-id-1"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    type="button"
                  >
                    <svg
                      aria-hidden={true}
                      aria-labelledby={null}
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style={
                        Object {
                          "verticalAlign": "-0.125em",
                        }
                      }
                      viewBox="0 0 192 512"
                      width="1em"
                    >
                      <path
                        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                      />
                    </svg>
                  </button>
                </div>
              </td>
            </tr>
            <tr
              className="pf-c-table__expandable-row"
              data-ouia-component-id="OUIA-Generated-TableRow-3"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe={true}
              hidden={true}
            >
              <td
                className=""
                colSpan={5}
                data-label="Content Details"
              >
                <div
                  className="pf-c-table__expandable-row-content"
                >
                  <span
                    aria-valuetext="Loading..."
                    className="pf-c-spinner pf-m-xl"
                    role="progressbar"
                  >
                    <span
                      className="pf-c-spinner__clipper"
                    />
                    <span
                      className="pf-c-spinner__lead-ball"
                    />
                    <span
                      className="pf-c-spinner__tail-ball"
                    />
                  </span>
                  <iframe
                    height="640"
                    hidden={true}
                    onLoad={[Function]}
                    title="Automated Analysis"
                    width="100%"
                  />
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      className="pf-c-drawer__panel pf-m-resizable"
      hidden={true}
      onTransitionEnd={[Function]}
    />
  </div>
</div>
`;
